web_src/js/features/repo-findfile.test.js

```diff
@@ -32,4 +32,9 @@ describe('Repo Find Files', () => {
     expect(res).toHaveLength(2);
     expect(res[0].matchResult).toEqual(['', 'word', '.txt']);
   });
+
+  test('escapePath', () => {
+    expect(escapePath('a/b/c')).toEqual('a/b/c');
+    expect(escapePath('a/b/ c')).toEqual('a/b/%20c');
```

Comment:
How about add a test for `#`?

Reply:
That's not necessary .... encodeURIComponent already does the right thing. Otherwise there will be a lot like `?`, `\`, etc .....

Reply:
From another perspective, if a users reports about `+` in file name doesn't work, do you want to test the `+`? And if another reports `%` doesn't work, do you want to test the `%`?

If you want to test `encodeURIComponent`, then every char execpt `-_~.` should be tested, I can not see the necessary.

**Test cases are designed for what it should be , but not what users said what happened.**

I can see many test cases in old code are not well designed, sometimes some developers just copy complex contents as test case data, which make the cases difficult to be maintained, and make future developers can not understand what the cases are for.

In a word: keep things simple and focused.